crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +37,32 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, _host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));           
+    pop!(interpreter, dest, src, len);
+    if len == U256::ZERO {
+        return;
+    }
+    let len = as_usize_or_fail!(interpreter, len, InstructionResult::InvalidOperandOOG);
+    gas_or_fail!(interpreter, gas::verylowcopy_cost(len as u64));
+
+
+    let dest = as_usize_or_fail!(
+        interpreter,
+        dest,
+        InstructionResult::InvalidOperandOOG
+    );
+    memory_resize!(interpreter, dest, len);
```

Comment:
I would assume that both dest and src can increment memory. So you need to find what is bigger and increment memory by a bigger value

Reply:
I made a function to check and handle this `memory_copy_resize` that was inspired by the geth pr's implementation in the `memory_table.go`